File: 2212-removing-minimum-and-maximum-from-array/removing-minimum-and-maximum-from-array.go
func minimumDeletions(nums []int) int {
    minIdx := 0
    maxIdx := 0
    n := len(nums) 
    for idx, val := range nums {
        if val < nums[minIdx] {
            minIdx = idx
        }
        if val > nums[maxIdx] {
            maxIdx = idx
        }
    }

    left := min(maxIdx, minIdx)
    right := max(maxIdx, minIdx)

    option1 := right + 1
    option2 := n - left
    option3 := (left + 1) + (n - right)

    return min(option1, min(option2, option3))
}

File: 2212-removing-minimum-and-maximum-from-array/removing-minimum-and-maximum-from-array.java
class Solution {
    public int minimumDeletions(int[] nums) {
        int minIdx = 0, maxIdx = 0, max = nums[0], min = nums[0];
        int n = nums.length;
        for(int i=0; i<n; i++) {
            if(nums[i] < min) {
                min = nums[i];
                minIdx = i;
            } 
            if(nums[i] > max) {
                max = nums[i];
                maxIdx = i;
            }
        }
        int left = Math.min(minIdx, maxIdx);
        int right = Math.max(minIdx, maxIdx);

        int option1 = right + 1;
        int option2 = n - left;
        int option3 = (left + 1) + (n - right);

        return Math.min(option1, Math.min(option2, option3));
    }
}